core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -107,36 +114,41 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
 
     //flip the workflows to Launching in a separate txn.
     //if this optimistic-lock-exceptions with another txn, this one will barf and we'll reschedule when we pipe it back to ourselves
-    workflowRecsToLaunch flatMap { wfRecs =>
-      dataSource.inTransaction { dataAccess =>
-        dataAccess.workflowQuery.batchUpdateStatus(wfRecs, WorkflowStatuses.Launching)
-      } map { count =>
-        if( wfRecs.nonEmpty ) {
-          SubmitWorkflowBatch(wfRecs.map(_.id))
-        } else {
-          ScheduleNextWorkflow
+    workflowRecsToLaunch flatMap {
+      case Some((wfRecs, submissionRec, workspaceRec)) if wfRecs.nonEmpty =>
+        dataSource.inTransaction { dataAccess =>
+          workflowStatusCounter(workspaceRec.toWorkspaceName, submissionRec.id, WorkflowStatuses.Launching).countDBResult {
+            dataAccess.workflowQuery.batchUpdateStatus(wfRecs, WorkflowStatuses.Launching)
+          } map { _ =>
+            SubmitWorkflowBatch(wfRecs.map(_.id), submissionRec, workspaceRec)
+          }
         }
-      }
+      case _ => Future.successful(ScheduleNextWorkflow)
     } recover {
       // if we found some but another actor reserved the first look again immediately
-      case t: RawlsConcurrentModificationException => ProcessNextWorkflow
+      case _: RawlsConcurrentModificationException => ProcessNextWorkflow
     }
   }
 
-  def reserveWorkflowBatch(dataAccess: DataAccess, activeCount: Int)(implicit executionContext: ExecutionContext): ReadWriteAction[Seq[WorkflowRecord]] = {
+  def reserveWorkflowBatch(dataAccess: DataAccess, activeCount: Int)(implicit executionContext: ExecutionContext): ReadWriteAction[Option[(Seq[WorkflowRecord], SubmissionRecord, WorkspaceRecord)]] = {
```

Comment:
+1 on Blergh. On the plus side it saves some database queries because we no longer need to re-lookup the `Workspace` and `Submission` in `submitWorkflowBatch` since we remember them from reservation time.

Maybe it could be made a bit cleaner with a container case class:
```
case class WorkflowBatch(workflowIds: Seq[Long], submissionRec: SubmissionRecord, workspaceRec: WorkspaceRecord)
```
and we pass that around instead of a `Tuple3`.